Signal/src/ViewControllers/MessagesViewController.m

```diff
@@ -500,6 +500,9 @@ - (void)viewWillAppear:(BOOL)animated
                                                                                                                @"Short name for edit menu item to share contents of media message.")
                                                                                       action:shareSelector],
                                                            ];
+
+    [self.collectionView.collectionViewLayout invalidateLayout];
```

Comment:
FYI - I wonder if `toggleObservers:` isn't responsible for some of these issues.